app/e2e/client/rocketchat.e2e.js

```diff
@@ -295,24 +297,24 @@ class E2E extends Emitter {
 	async requestPassword() {
 		return new Promise((resolve) => {
 			const showModal = () => {
-				modal.open({
-					title: TAPi18n.__('Enter_E2E_password_to_decode_your_key'),
-					type: 'input',
-					inputType: 'password',
-					html: true,
-					text: `<div>${ TAPi18n.__('E2E_password_request_text') }</div>`,
-					showConfirmButton: true,
-					showCancelButton: true,
-					confirmButtonText: TAPi18n.__('Decode_Key'),
-					cancelButtonText: TAPi18n.__('I_ll_do_it_later'),
-				}, (password) => {
-					if (password) {
-						this.closeAlert();
-						resolve(password);
-					}
-				}, () => {
-					failedToDecodeKey = false;
-					this.closeAlert();
+				imperativeModal.open({ component: EnterE2EPasswordModal,
```

Comment:
Since you created a modal in rect for this specific use, I believe it's better to pass it's specific props inside the modal (such as title, variant, cancelText, confirmText).